zcash_client_backend/src/data_api.rs

```diff
@@ -496,6 +500,51 @@ impl<A: Copy> Account for (A, UnifiedIncomingViewingKey) {
     }
 }
 
+/// Information about an address in the wallet.
+pub struct AddressInfo {
+    address: Address,
+    diversifier_index: DiversifierIndex,
+    #[cfg(feature = "transparent-inputs")]
+    transparent_key_scope: Option<TransparentKeyScope>,
+}
+
+impl AddressInfo {
+    /// Constructs an `AddressInfo` from its constituent parts.
+    pub fn from_parts(
+        address: Address,
+        diversifier_index: DiversifierIndex,
+        #[cfg(feature = "transparent-inputs")] transparent_key_scope: Option<TransparentKeyScope>,
+    ) -> Self {
+        Self {
+            address,
+            diversifier_index,
+            #[cfg(feature = "transparent-inputs")]
+            transparent_key_scope,
+        }
+    }
+
+    /// Returns the address this information is about.
+    pub fn address(&self) -> &Address {
+        &self.address
+    }
+
+    /// Returns the diversifier index the address was derived at.
+    pub fn diversifier_index(&self) -> DiversifierIndex {
+        self.diversifier_index
+    }
+
+    /// Returns the key scope if this is a transparent address.
+    #[cfg(feature = "transparent-inputs")]
+    pub fn transparent_key_scope(&self) -> Option<TransparentKeyScope> {
+        // Internally match to ensure we ignore `self.transparent_key_scope` if it was set
+        // to nonsense.
```

Comment:
This should be checked in `from_parts` (in which case it does not need to be checked here, because the fields are private and it will be an enforced invariant).

Reply:
*sigh* I knew someone would ask for that